src/goose/brokers/redis/batch.clj

```diff
@@ -0,0 +1,25 @@
+(ns goose.brokers.redis.batch
+  (:require [goose.brokers.redis.commands :as redis-cmds]
+            [goose.defaults :as d]
+            [taoensso.carmine :as car]))
+
+(defn- set-batch-state
+  [{:keys [id callback-fn-sym jobs]}]
+  (let [batch (d/prefix-batch id)
+        enqueued-jobs-set (d/suffix-batch-set id d/enqueued-job-set)]
+    (car/hmset batch
+               (name :id) id
+               (name :callback-fn-sym) callback-fn-sym)
+    (apply #(car/sadd enqueued-jobs-set (:id %)) jobs)))
+
+(defn- enqueue-jobs
+  [jobs]
+  (doseq [job jobs]
+    (car/lpush (:ready-queue job) job)))
```

Comment:
if the job is of cron type, does this line still work? Or are we assuming that all jobs in this batch will be for ready queue only.
coz perform-async and perform-in-sec are different functionalities.

Reply:
We need to call this assumption out more clearly

Reply:
As discussed, renamed the client fn to `perform-batch-async` to be more explicit about this assumption, for now.

Reply:
Sorry, reopening this conversation. Users don't submit jobs to a batch, only function & list of args. We shouldn't rename `perform-batch`. 

If we must, we can rename batch's key from `:job` to `:async-jobs`. Although, I find this unnecessary. `(:ready-queue job)` in the context of batch namespace is not confusing to me.

Reply:
Undoing the rename of the fn. And adding a docstring that makes it clear that the batch jobs will be scheduled for immediate execution on the client fn.